core/include/drivers/sp805_wdt.h

```diff
@@ -0,0 +1,67 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2019 Broadcom.
+ */
+
+#ifndef SP805_H
+#define SP805_H
+
+#include <drivers/wdt.h>
+#include <kernel/interrupt.h>
+#include <types_ext.h>
+
+/* SP805 register offset */
+#define WDT_LOAD_OFFSET		0x000
+#define WDT_CONTROL_OFFSET	0x008
+#define WDT_INTCLR_OFFSET	0x00c
+#define WDT_LOCK_OFFSET		0xc00
+
+/* Magic word to unlock the wd registers */
+#define WDT_UNLOCK_KEY		0x1ACCE551
+#define WDT_LOCK_KEY		0x1
+
+/* Register field definitions */
+#define WDT_INT_EN		BIT(0)
+#define WDT_RESET_EN		BIT(1)
+#define WDT_INT_CLR		BIT(0)
+
+#define WDT_LOAD_MIN		0x1
+#define WDT_LOAD_MAX		0xffffffff
+
+typedef enum itr_return (*irq_handler_t)(struct itr_handler *h __unused);
+
+struct sp805_wdt_data {
+	struct io_pa_va base;
+	struct wdt_chip chip;
+	uint32_t clk_rate;
+	uint32_t load_val;
+	uint32_t irq_num;
```

Comment:
irq is a bit confusing since it depends on interrupts controller (GICv2 vs GICv3 for instance) if secure interrupts are delivered as IRQ or FIQ.

Reply:
will replace irq by itr.